src/auth/session/storage/memory.ts

```diff
@@ -19,4 +19,18 @@ export class MemorySessionStorage implements SessionStorage {
     }
     return true;
   }
+
+  public async deleteSessions(ids: string[]): Promise<boolean> {
+    ids.forEach((id) => delete this.sessions[id]);
+    return true;
+  }
+
+  public async findSessionsByShop(
+    shop: string,
+  ): Promise<SessionInterface[] | {[key: string]: unknown}[] | undefined> {
+    const results = Object.values(this.sessions).filter(
+      (session) => session.shop === shop,
+    );
+    return results.length === 0 ? undefined : results;
```

Comment:
The `undefined` return here I think makes it a bit awkward - could we just always return `results`, even if it's empty?